### PR TITLE
fix: Avoid app crash on saving after screen rotation

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- App crashes because when the screen is rotated, the recycler view is destroyed. This is avoided by setting retainInstance to true in the onCreate() of the fragment
- Changing same id name for different recyclerView
- Currently, saving a configuration is in onPostExecute() of the AsyncTask class, which means that action saving is done in the main thread. I put it back into the doInBackground()

Fixes: #211

### CODE REVIEW

**Comment by anhanh11001** [comment]
Actually, to fix this issue, I only need to put these 4 lines of code. Others are just changes to improve the codebase.
```
override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     retainInstance = true
}
```
From what I understand, your suggestion means that if recycler_view is null then we don't update the list at all, which I think is more like avoiding the issue, not really fixing it.

**Comment by adityastic** [comment]
> Actually, to fix this issue, I only need to put these 4 lines of code. Others are just changes to improve the codebase.
> 
> ```
> override fun onCreate(savedInstanceState: Bundle?) {
>      super.onCreate(savedInstanceState)
>      retainInstance = true
> }
> ```
> From what I understand, your suggestion means that if recycler_view is null then we don't update the list at all, which I think is more like avoiding the issue, not really fixing it.

So you're just retaining the fragment which will avoid it from being recreated. Cool.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt [hunk 1]
```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
         return inflater.inflate(R.layout.fragment_main_text, container, false)
     }
 
-    class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
+    inner class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
         override fun doInBackground(vararg params: Void?): Boolean? {
-            return (!StorageUtils.checkIfFilePresent(filename))
+            return when (StorageUtils.checkIfFilePresent(filename)) {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: why did you change it?

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt [hunk 2]
```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
...
+                true -> {
+                    saveFile()
+                    true
+                }
+                false -> showFileOverrideDialog()
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: very vague, wont work.

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt [hunk 3]
```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
...
         }
 
         private fun saveFile() {
             StorageUtils.saveFile(filename, json)
-            listener.updateSavedList()
         }
 
-        private fun showFileOverrideDialog() {
+        private fun showFileOverrideDialog(): Boolean {
+            var saveSuccess = false
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: Why are you initialising new variables?

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt [hunk 4]
```diff
@@ -314,35 +314,41 @@ class MainTextDrawableFragment : BaseFragment() {
         return inflater.inflate(R.layout.fragment_main_text, container, false)
     }
 
-    class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
+    inner class StoreAsync(private val filename: String, private val json: String, private val context: WeakReference<Context>, private val listener: PreviewChangeListener) : AsyncTask<Void, Void, Boolean>() {
         override fun doInBackground(vararg params: Void?): Boolean? {
-            return (!StorageUtils.checkIfFilePresent(filename))
+            return when (StorageUtils.checkIfFilePresent(filename)) {
+                true -> {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: Please revert StoreAsync to its original state. This is messed up, as far as I know this logic will not work

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt (hunk 2 above)
Comment (anhanh11001) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: From what I understand, you made the StoreAsync class to make the action of saving a new configuration in a different thread. Is it correct?

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: yep correct

Comment (anhanh11001) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: well then currently, the function saveFile() is called in onPostExecute(), which is in the main thread. What I did was to move in into the different thread and then check if action saving is successful then   call listener.updateSavedList()

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: yeah you're completely right on that. But you're calling `showFileOverrideDialog()` in doInBackground. It just wont work, it is not in the main thread and hence will not have the application context with it

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: and also, the logic is the opposite, you should save the file if it did not exist. It is the opposite right now. Have a look there

Comment (anhanh11001) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: oh right that's true, I'll send update later

File: app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt [hunk 5]
```diff
@@ -61,71 +60,8 @@ class MainTextDrawableFragment : BaseFragment() {
         }
     }
 
-    override fun getSendData(): DataToSend {
```
Comment (adityastic) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: It is removed and copied at the bottom. Please fix this

Comment (anhanh11001) [reply app/src/main/java/com/nilhcem/blenamebadge/ui/fragments/MainTextDrawableFragment.kt]: Fixed
